src/handler/handlers.jl

```diff
@@ -162,7 +163,47 @@ function process_assets(request::HTTP.Request, state::HandlerState; file_path::A
     
 end
 
-const dash_router = HTTP.Router()
+function process_reload_hash(request::HTTP.Request, state::HandlerState)
+    reload_tuple = (
+        reloadHash = state.reload.hash,
+        hard = state.reload.hard,
+        packages = keys(state.cache.resources.files),
+        files = state.reload.changed_assets
+    ) 
+    state.reload.hard = false
+    state.reload.changed_assets = []
+    return HTTP.Response(200, ["Content-Type" => "application/json"], body = JSON2.write(reload_tuple))
+
+end
+
+function start_reaload_poll(state::HandlerState)
+    folders = Set{String}()
+    push!(folders, get_assets_path(state.app))
+    push!(folders, state.registry.dash_renderer.path)
+    for pkg in values(state.registry.components)
+        push!(folders, pkg.path)
+    end
+    state.reload.task = @async poll_folders(folders; interval = get_devsetting(state.app, :hot_reload_watch_interval)) do file, ts, deleted
```

Comment:
@waralex Is this `poll_folders` call run using `@async` because it's potentially blocking otherwise? I see that it's a `while` loop, so it runs continuously while hot reloading is active, including a sleep interval.

I'm guessing that the thread initiated by the `@async` call is terminated when the Julia process that spawned it terminates also. Since we kill the Julia process after spawning the subprocess in a hot reloading event, I'm hopeful this means that all background activity for a given app session ends at the same time.